File: C++/A1/049.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
using namespace std;
int main(){
    string original,reversed;
    cin >> original;
    reversed = original;
    reverse(reversed.begin(),reversed.end());
    int d1 = original[0]-'0',d2 = original[1]-'0',d3 = original[2]-'0',d4 = original[3]-'0',d5 = original[4]-'0',sum=d1+d2+d3+d4+d5,multiply=d1*d2*d3*d4*d5;
    if(d1>5)cout<<9;
    else if(d2>5)cout<<10;
    else if(d3>5)cout<<11;
    else if(d4>5)cout<<12;
    else if(d5>5)cout<<14;
    else cout << 13;

    if(original == reversed){
        if(d1+d5>5)cout<<1;
        else if(d2*d4>5)cout<<2;
        else cout<<0;
    }else{
        if(round((double)d5 / d1) > 5)cout<<1;
        else if(d2-d5>5)cout<<2;
        else cout<<0;
    }
    if(sum>25)cout<<1;
    else if(multiply>55)cout<<2;
    else cout<<0;
}

File: Python/A1/049.py
A = input()
result = ""
n = [int(A[0]),int(A[1]),int(A[2]),int(A[3]),int(A[4])]
if(n[0] > 5): result += "9"
elif(n[1] > 5): result += "10"
elif(n[2] > 5): result += "11"
elif(n[3] > 5): result += "12"
elif(n[4] > 5): result += "14"
else: result += "13"

if(A==A[::-1]):
    if(n[0]+n[4]>5): result += "1"
    elif(n[1]*n[3]>5): result += "2"
    else: result += "0"
else:
    if(n[4]//n[1]>5): result += "1"
    elif(n[1]-n[4]>5): result += "2"
    else: result += "0"
if(sum(n)>25): result += "1"
elif(n[0]*n[1]*n[2]*n[3]*n[4]>55): result += "2"
else: result += "0"
print(result)
